### DISCUSSION TITLE
Applying Deep Learning to Autonomous Driving

### DISCUSSION THREAD

**Discussion by MariaPiaGelos** [prl-mushr/mushr#54] [General]
I want to implement your MuSHR DL model on a real donkey car hardware.

Has anyone tried it? What should I modify on the framework?

**Comment by sidtalia** [comment]
Hi,

A similar question has been asked and answered [here](https://github.com/prl-mushr/mushr/discussions/52), you could reach out to the person that asked the question and see if they made any progress. TL;DR, we haven't tried it on the real car's hardware. The purpose of the framework was to give a feel for DL to people that were already using MuSHR cars in some manner. 

One could use the DL framework to test an idea in simulation (say a different network architecture). As for possible modifications to use the trained model on the donkey car, as long as the image dimensions match the dimensions expected by the network, it should work in principle. However, Since the framework only predicts steering angle, throttle control would have to be created as well. We haven't implemented it on the actual car yet so we don't know the exact challenges that would be faced.

As for implementing it on the donkey-car hardware, you would have to write your own code for the hardware abstraction stuff since the hardware of the two cars is quite different.
